### PR TITLE
literature: fix for ``earliest_date``

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: inspirehep/records/api/literature.py [hunk 1]
```diff
@@ -77,18 +77,14 @@ def earliest_date(self):
         dates = [
             str(el)
             for el in chain.from_iterable(
-                force_list(self.get_value(path)) for path in date_paths
+                [force_list(self.get_value(path)) for path in date_paths]
```
Comment (michamos) [inline inspirehep/records/api/literature.py]: you don't need to do this, as it can take any iterable.

File: inspirehep/records/api/literature.py [hunk 2]
```diff
@@ -77,18 +77,14 @@ def earliest_date(self):
...
             )
         ]
-        proper_dates = []
-        for date in dates:
-            try:
-                proper_dates.append(dateutil.parser.parse(date))
-            except ValueError:
-                pass
-        if proper_dates:
-            date = earliest_date(dates, full_date=True)
```
Comment (michamos) [inline inspirehep/records/api/literature.py]: the `full_date` is needed for the citation count table, I don't think it matters for ES.

File: tests/integration-async/records/indexer/test_indexer_cli.py [hunk 3]
```diff
@@ -82,7 +82,7 @@ def test_reindex_record_lit_fails_with_invalid_field_content(
     generate_records,
     check_n_records_reindex_for_pidtype,
 ):
-    invalid_field = {"preprint_date": "i am not a date"}
+    invalid_field = {"titles": ["i am not a date"]}
```
Comment (michamos) [inline tests/integration-async/records/indexer/test_indexer_cli.py]: maybe change the content of the string? `I am not an object`

File: tests/integration-async/records/indexer/test_indexer_cli.py [hunk 4]
```diff
@@ -98,7 +98,7 @@ def test_reindex_records_lit_one_fails_and_two_ok(
...
```
Comment (michamos) [inline tests/integration-async/records/indexer/test_indexer_cli.py]: same

**Comment by drjova** [comment]
👍 done

**Pull request merged**